gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -1161,6 +1201,124 @@ public Operation deprecate(ImageId image,
     }
   }
 
+  @Override
+  public Disk get(final DiskId diskId, DiskOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Disk answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Disk>() {
+            @Override
+            public com.google.api.services.compute.model.Disk call() {
+              return computeRpc.getDisk(diskId.zone(), diskId.disk(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Disk.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Operation create(final DiskInfo disk, OperationOption... options) {
+    final com.google.api.services.compute.model.Disk diskPb =
+        disk.setProjectId(options().projectId()).toPb();
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      return Operation.fromPb(this,
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.createDisk(disk.diskId().zone(), diskPb, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER));
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  private static Function<com.google.api.services.compute.model.Disk, Disk> diskFromPb(
```

Comment:
The function in `Blob` takes a tuple and is only used in `BatchResponse submit(BatchRequest batchRequest)` to allow `transformBatchResult` to be more generic. The function takes a tuple, if we have a collection I think it will take more (both time and memory) to create a tuple `<Strorage, Disk>` for each item in the collection and then call the `Function` rather than having a factory method for the transformation function which we only need to call once for each collection transformation.